k2/csrc/weights.h

```diff
@@ -29,53 +34,52 @@ namespace k2 {
   but with the opposite sign, as in logprobs rather than negative logprobs.
   It's like shortest path but with the opposite sign.
 
-   @param [in] fsa  The fsa we are doing the forward computation on.
-                    Must satisfy IsValid(fsa) and IsTopSorted(fsa).
+   @param [in]  fsa  The fsa we are doing the forward computation on.
+                Must satisfy IsValid(fsa) and IsTopSorted(fsa).
+   @param [in]  arc_weights  Arc weights, indexed by arc in `fsa`.
+                             Usually logprobs.
    @param [out] state_weights  The per-state weights will be written to here.
                 They will be 0 for the start-state (if fsa is
                 nonempty), and for later states will be the
                 largest (most positive) weight from the start-state
-                to that state along any path, or +infinity if no such
+                to that state along any path, or `kNegativeInfinity` if no such
                 path exists.
  */
-void ComputeForwardMaxWeights(const Fsa &fsa, float *state_weights);
+void ComputeForwardMaxWeights(const Fsa &fsa, const float *arc_weights,
+                              float *state_weights);
```

Comment:
Why is the argument `state_weights` of type `float`, but the members are of type `double`?

Reply:
`ComputeForwardMaxWeights` is not a member of `WfsaWithFbWeights`.

Reply:
What I wanted to express is that both the arguments of this function and the data members of  `WfsaWithFbWeights` represent **state** weight, but the function arguments are `float`
but the data members are **double**.

Why not to change the arguments to type `double` ?